src/main/ruby/core/transcoding.rb

```diff
@@ -119,10 +119,12 @@ def self.asciicompat_encoding(string_or_encoding)
       return unless encoding
       return if encoding.ascii_compatible?
 
-      transcoding = TranscodingMap[encoding.name.upcase.to_sym]
-      return unless transcoding and transcoding.size == 1
+      name = encoding.name.upcase.to_sym
+      transcoders = Truffle.invoke_primitive :encoding_transcoders_from_encoding, name
 
-      Encoding.find transcoding.keys.first.to_s
+      return unless transcoders and transcoders.size == 1
+
+      Encoding.find transcoders[0].to_s
```

Comment:
Minor, but a lot of this code uses `.first` instead of `[0]`.

Reply:
I find `[0]` shorter, but if you think `first` is clearer here I can change.
(There really should be a Enumerable#single/only or so)